.github/workflows/build-mac.sh

```diff
@@ -56,7 +56,7 @@ function build() {
   echoRun mkdir -p zipline/src/jvmMain/resources/jni/$MACOS_ARCH/
   echoRun cmake -S zipline/src/jvmMain/ -B build/jni/$MACOS_ARCH/ \
     -DQUICKJS_VERSION="$(cat zipline/native/quickjs/VERSION)" \
-    -DCMAKE_SYSTEM_PROCESSOR=$CMAKE_ARCH
+    -DCMAKE_OSX_ARCHITECTURES=$CMAKE_ARCH
```

Comment:
I determined experimentally that this is the variable I want to change.
https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html
https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html